gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClient.java

```diff
@@ -0,0 +1,350 @@
+  }
+
+  /**
+   * Get the import links of the input URI.
+   * 
+   * @param configKeyUri - The URI for the configuration key. 
+   * @param recursive    - Specify whether to get direct import links or recursively import links
+   * @return  the import links of the input URI.
+   * 
+   * @throws ConfigStoreFactoryDoesNotExistsException: if missing scheme name or the scheme name is invalid
+   * @throws ConfigStoreCreationException: Specified {@link ConfigStoreFactory} can not create required {@link ConfigStore}
+   * @throws VersionDoesNotExistException: Required version does not exist anymore ( may get deleted by retention job )
+   */
+  public Collection<URI> getImports(URI configKeyUri, boolean recursive)
+      throws ConfigStoreFactoryDoesNotExistsException, ConfigStoreCreationException, VersionDoesNotExistException {
+    ConfigStoreAccessor accessor = this.getConfigStoreAccessor(configKeyUri);
+    ConfigKeyPath configKeypath = ConfigClientUtils.buildConfigKeyPath(configKeyUri, accessor.configStore);
+    Collection<ConfigKeyPath> result;
+    
+    if(!recursive){
+      result = accessor.topologyInspector.getOwnImports(configKeypath);
+    }
+    else{
+      result = accessor.topologyInspector.getImportsRecursively(configKeypath);
+    }
+    
+    return ConfigClientUtils.getAbsoluteURI(result, accessor.configStore);
+  }
+
+  /**
+   * Get the URIs which imports the input URI
+   * 
+   * @param configKeyUri - The URI for the configuration key. 
+   * @param recursive    - Specify whether to get direct or recursively imported by links
+   * @return  the URIs which imports the input URI
+   * 
+   * @throws ConfigStoreFactoryDoesNotExistsException: if missing scheme name or the scheme name is invalid
+   * @throws ConfigStoreCreationException: Specified {@link ConfigStoreFactory} can not create required {@link ConfigStore}
+   * @throws VersionDoesNotExistException: Required version does not exist anymore ( may get deleted by retention job )
+   */
+  public Collection<URI> getImportedBy(URI configKeyUri, boolean recursive)
+      throws ConfigStoreFactoryDoesNotExistsException, ConfigStoreCreationException, VersionDoesNotExistException {
+    ConfigStoreAccessor accessor = this.getConfigStoreAccessor(configKeyUri);
+    ConfigKeyPath configKeypath = ConfigClientUtils.buildConfigKeyPath(configKeyUri, accessor.configStore);
+    Collection<ConfigKeyPath> result;
+    
+    if(!recursive){
+      result = accessor.topologyInspector.getImportedBy(configKeypath);
+    }
+    else{
+      result = accessor.topologyInspector.getImportedByRecursively(configKeypath);
+    }
+    
+    return ConfigClientUtils.getAbsoluteURI(result, accessor.configStore);
+  }
+  
+  private URI getMatchedFloorKeyFromCache(URI configKeyURI){
+    URI floorKey = this.configStoreAccessorMap.floorKey(configKeyURI);
+    if(floorKey==null) {
+      return null;
+    }
+    
+    // both scheme name and authority name should match
```

Comment:
This is not correct. If the authority of configKeyURI is null, then only the scheme should match.